### PR TITLE
Fixture: Legacy JSX Runtimes

### PR DESCRIPTION
**Pull request by gaearon**
Adds checked-in bundles for JSX runtimes for different stable React versions. Those were edited by hand.
Each comes with an integration testing suite adjusted for that version.

They look like this:

Screenshot 2020-10-14 at 00 16 37

```
cd fixtures/legacy-jsx-runtimes 
yarn
yarn test
```

Screenshot 2020-10-14 at 00 17 23

The regression suite contains a lint step and some hand-picked tests adapted from different existing ones in the repo. We run in three configurations: `jsxDEV in DEV`, `jsx in DEV`, and `jsx in PROD`.

## Differences Between Bundles

Each lower bundle is based on the higher one.

- 16: Use old stacks. Remove `object-assign` from the output.
- 15: Read old stacks from `ReactComponentTreeHook`.
- 14: Remove stacks. Use `instance.getName()` in messages. Read `ReactCurrentOwner` from `react/lib/`.

### CODE REVIEW

File: fixtures/legacy-jsx-runtimes/react-16/react-16.test.js [hunk 1]
```diff
@@ -0,0 +1,769 @@
...
+};
+let RequiredPropComponent = class extends React.Component {
+  render() {
+    return <span>{this.props.prop}</span>;
+  }
+};
+RequiredPropComponent.displayName = 'RequiredPropComponent';
+RequiredPropComponent.propTypes = {prop: PropTypes.string.isRequired};
+
+it('works', () => {
```
Comment (rickhanlonii) [inline fixtures/legacy-jsx-runtimes/react-16/react-16.test.js:37]: Math checks out

File: fixtures/legacy-jsx-runtimes/react-15/react-15.test.js [hunk 2]
```diff
@@ -0,0 +1,771 @@
+/**
```
Comment (rickhanlonii) [inline fixtures/legacy-jsx-runtimes/react-15/react-15.test.js:1]: May want to reference `ReactJSXElement-test` in a comment so it's clear where these come from (and we may want to add a comment in `ReactJSXElement-test` to remind us to update these tests.

File: fixtures/legacy-jsx-runtimes/setupTests.js [hunk 3]
```diff
@@ -0,0 +1,315 @@
+const expect = global.expect;
```
Comment (rickhanlonii) [inline fixtures/legacy-jsx-runtimes/setupTests.js:1]: Same here, a comment explaining that this is duplicated from elsewhere.

**Review by rickhanlonii — APPROVED** [review]
Great stuff!!

File: fixtures/legacy-jsx-runtimes/react-15/react-15.test.js (hunk 2 above)
Comment (gaearon) [reply fixtures/legacy-jsx-runtimes/react-15/react-15.test.js:1]: It’s not just that file but some from a bunch of others too. I could add though.

**Pull request merged**